cmd/atlas/internal/cmdapi/migrate.go

```diff
@@ -525,6 +535,36 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
 	}
 }
 
+func revisionsTableExists(ctx context.Context, c *sqlclient.Client) (bool, error) {
+	// Connect to the given schema name.
+	sc, err := sqlclient.Open(ctx, MigrateFlags.URL, sqlclient.OpenSchema(MigrateFlags.RevisionSchema))
+	switch {
+	case err != nil && !errors.Is(err, sqlclient.ErrUnsupported):
+		return false, err
+	case errors.Is(err, sqlclient.ErrUnsupported):
+		// If the driver does not support changing the schema use the existing connection.
+		sc = c
+	case err == nil:
+		// Connecting attempt to the schema was successful, make sure to close it.
+		defer sc.Close()
+	}
+	// Inspect schema and check if the table does already exist.
+	s, err := sc.InspectSchema(ctx, "", &schema.InspectOptions{Tables: []string{revision.Table}})
+	switch {
+	case err != nil && !schema.IsNotExistError(err):
+		return false, err
+	case schema.IsNotExistError(err):
+		// Schema does not exist.
+		return false, nil
+	}
```

Comment:
```suggestion
	if err != nil {
		if schema.IsNotExistError(err) {
			err = nil
		}
		return false, err
	}
```
More level of nesting, but feels simpler to read it this way imo.